tests/area/test_markets.py

```diff
@@ -0,0 +1,132 @@
+    def test_market_rotation_is_successful_keep_past_markets(self, area_fixture):
+        area_fixture.activate()
+        assert len(area_fixture.all_markets) == 5
+        slot_length_in_ticks = area_fixture.config.slot_length / area_fixture.config.tick_length
+        area_fixture.current_tick = slot_length_in_ticks
+        area_fixture.cycle_markets()
+        assert len(area_fixture.past_markets) == 1
+        assert len(area_fixture.all_markets) == 5
+
+        area_fixture.current_tick = slot_length_in_ticks * 2
+        area_fixture.cycle_markets()
+        assert len(area_fixture.past_markets) == 2
+        assert len(area_fixture.all_markets) == 5
+
+    @patch("d3a_interface.constants_limits.ConstSettings.BalancingSettings."
+           "ENABLE_BALANCING_MARKET", True)
+    def test_balancing_market_rotation_is_successful(self, area_fixture):
+        area_fixture.activate()
+        assert len(area_fixture.balancing_markets) == 5
+        slot_length_in_ticks = area_fixture.config.slot_length / area_fixture.config.tick_length
+        area_fixture.current_tick = slot_length_in_ticks
+        area_fixture.cycle_markets()
+        assert len(area_fixture.past_balancing_markets) == 1
+        assert len(area_fixture.balancing_markets) == 5
+
+        area_fixture.current_tick = slot_length_in_ticks * 2
+        area_fixture.cycle_markets()
+        assert len(area_fixture.past_balancing_markets) == 1
+        assert len(area_fixture.balancing_markets) == 5
+
+    @patch("d3a_interface.constants_limits.ConstSettings.GeneralSettings."
+           "ENABLE_SETTLEMENT_MARKETS", True)
+    def test_settlement_market_rotation_is_successful(self, area_fixture):
+        """
+        #slot   #markets #past_markets #settlement_markets #past_settlement_markets
+        1            5         1             1                   0
+        2            5         2             2                   0
+        3            5         3             3                   0
+        4            5         4             4                   0
+        5            5         4             4                   1
+        6            5         4             4                   1
+
+        """
+        expected_market_cycles = {
+            area_fixture.config.start_date: [5, 1, 1, 0],
+            area_fixture.config.start_date + 1 * area_fixture.config.slot_length: [5, 2, 2, 0],
+            area_fixture.config.start_date + 2 * area_fixture.config.slot_length: [5, 3, 3, 0],
+            area_fixture.config.start_date + 3 * area_fixture.config.slot_length: [5, 4, 4, 0],
+            area_fixture.config.start_date + 4 * area_fixture.config.slot_length: [5, 4, 4, 1],
+            area_fixture.config.start_date + 5 * area_fixture.config.slot_length: [5, 4, 4, 1],
+        }
+
+        area_fixture.activate()
+        slot_length_in_ticks = area_fixture.config.slot_length / area_fixture.config.tick_length
```

Comment:
```suggestion
        total_ticks_per_slot = area_fixture.config.slot_length / area_fixture.config.tick_length
```
NIT: Sounds clearer to me what it is

Reply:
I response to [Hassans comment](https://github.com/gridsingularity/d3a/pull/1213#discussion_r698360099) I renamed it to `ticks_per_slot` OK?